### DISCUSSION TITLE
Unable to find the pattern - issuer=C=US,O=Network Solutions L.L.C.,CN=Network Solutions Certificate Authority in OpenSSL 3.0.11

### DISCUSSION THREAD

**Discussion by debananda-pal** [openssl/openssl#25141] [Q&A]
Hi,

We have moved from Buster to Bookworm and using OpenSSL 3.0.13.
I am getting below issue while installing CRL certificate cert-name.crl.pem  :

FAIL	testFailed: Unable to find the pattern - issuer=C=US,O=Network Solutions L.L.C.,CN=Network Solutions Certificate Authority

Please suggest if there is any change in OpenSSL 3.0.13.

Thanks,
D Pal

**Comment by mattcaswell** [comment]
What is the exact openssl command that you are using?

**Reply by debananda-pal** [reply]
Thank you mattcaswell.
We are not using openssl command, but getting error as below :
"Error getting CRL issuer certificate"

I guess above error message is coming form "src/openssl/apps/crl.c"

Thanks

**Reply by mattcaswell** [reply]
> I guess above error message is coming form "src/openssl/apps/crl.c"

That would make sense since that code does emit an error message of exactly that form. However you say "We are not using openssl command" - but that code file is part of the source code for the openssl command line application.

So how are your doing this? Are you using some script that in turn is calling the openssl command line?

**Comment by debananda-pal** [comment]
Hi mattcaswell,

The below command is executed to get the issuer details :
/usr/bin/openssl crl -verify -noout -nameopt sep_comma_plus -issuer -lastupdate -nextupdate -inform=pem -CAfile /config/certs/store/ca-certificates.crt -in /home/admin/Cert.crl.pem

But above command didn't return issuer details and installation  succeeded with error as "Error getting CRL issuer certificate".

**Comment by debananda-pal** [comment]
Hi mattcaswell,
Please see few findings below, after debugging the code :
1. In function x509_object_idx_cnt(), X509_LOOKUP_TYPE is passed as X509_LU_X509
2. In function x509_object_idx_cnt(), sk_X509_OBJECT_find_all() returns "-1"
3. In function X509_STORE_CTX_get_by_subject(), X509_OBJECT_retrieve_by_subject() returns NULL
4. In function X509_OBJECT_retrieve_by_subject(), X509_OBJECT_idx_by_subject() returns "-1".

I am using OpenSSL 3.0.11 debian package. Please let me know your suggestion.

Regards,
D Pal

**Comment by debananda-pal** [comment]
Hi mattcaswell,
If I remove -verify and -CAfile option (shown below), I am not getting the above Error.
/usr/bin/openssl crl -noout -nameopt sep_comma_plus -issuer -lastupdate -nextupdate -inform=pem -in /home/admin/Cert.crl.pem

Please let me know that with removal of -verify and -CAfile options, the installed certificate will work seamlessly.

Thanks,
D Pal
